core/src/main/scala/kafka/log/Log.scala

```diff
@@ -68,6 +68,13 @@ object LogAppendInfo {
       offsetsMonotonic = false, -1L, recordErrors, errorMessage)
 }
 
+sealed trait LeaderHWChange
+object LeaderHWChange {
+  case object Incremental extends LeaderHWChange
```

Comment:
Probably Increased is clearer than Incremental.

Reply:
Another nit: `LeaderHwChange` adheres to the coding convention better.